shared_web/static/js/pd.js

```diff
@@ -367,6 +368,70 @@ PD.htmlEscape = function (s) {
     return $("<div>").text(s).html();
 };
 
+PD.Filter = {};
+
+PD.Filter.init = function () {
+    // Apply the filter with the initial value of the form
+    PD.Filter.scryfallFilter($("#scryfall-filter-input").val())
+
+    // set up the event handlers for the form
+    $("#scryfall-filter-form").submit(function () {
+        PD.Filter.scryfallFilter($("#scryfall-filter-input").val());
+        return false;
+    });
+    $("#scryfall-filter-reset").click(PD.Filter.reset);
+};
+
+PD.Filter.scryfallFilter = function (query) {
+    if (query == ""){
+        PD.Filter.reset();
+        return;
+    }
+
+    $("#scryfall-filter-submit").attr("disabled", "disabled").text("Loading...");
+    $("#scryfall-filter-reset").attr("disabled", "disabled").text("Loading...");
+    $("#scryfall-filter-form").submit(function () { return false; })
+
+    query = "f:pd and (" + query + ")";
+    card_names = [];
+
+    function doFilter () {
+        $("tr.cardrow").each( function () {
+            jqEle = $(this)
+            if (card_names.indexOf(this.id.split('-')[1]) == -1){
+                jqEle.hide();
+            }
+            else {
+                jqEle.show();
+            }
+        });
+        $("#scryfall-filter-submit").removeAttr("disabled").text("Search");
+        $("#scryfall-filter-reset").removeAttr("disabled").text("Reset");
+        $("#scryfall-filter-form").submit(function () { return false; })
+    }
+
+    function parse_and_continue (blob) {
+        // TODO: error handling
+
+        for (i=0; i<blob.data.length; i++) {
+            card_names.push(blob["data"][i]["name"])
+        }
+        if (blob["has_more"]) {
+            $.getJSON(blob["next_page"], parse_and_continue)
+        }
+        else {
+            doFilter();
+        }
+    }
+
+    $.getJSON("https://api.scryfall.com/cards/search?q=" + query, parse_and_continue);
+
+};
+
+PD.Filter.reset = function () {
+    $("tr.cardrow").show();
+};
+
```

Comment:
If we're going to expand this kind of filtering to the cards page we should make it consistent with the filtering on /rotation/. Either they both use javascript or neither do.

How is the performance?

Reply:
I think the performance is basically same, it's gated by making the requests to scryfall and parsing all the different pages. This way, the requests are directly between the user and Scryfall, rather than between prod and Scryfall - so maybe faster or slower. The old way, the whole page has to be reloaded, but this way, jQuery has to go through the page modifying the DOM. Also, doing it with JS means there's no extra hits to the database, but those hits aren't that expensive since it's all coming from cached queries.

I personally think doing it in JS is snazzier and I envisioned it being nice for future parts of this page, like e.g. if I can make a search for cards fitting a certain Scryfall fliter (like the colors of my deck) and then do different sub-filters like cards good vs archetype 1, cards good vs archetype 2. But I don't feel strongly. I do agree that /cards/ and /rotation/ should be unified, though.

Reply:
I think for specific queries it can be faster with JS. Something like `t:creature` or `c:u` will be about the same, but something like `t:land` is probably quicker with JS.

Reply:
I'm not against a js version. It seems like it will probably be a significantly better experience as long as we handle super slow queries/slow connections well. I'm not sure how I feel about querying the scryfall API directly and not via `fetcher`. I guess it's ok/better than going an additional hop through some endpoint on pdm. Not sure about that. How hard to swap this in on /rotation/ too?